src/lib.rs

```diff
@@ -1,4 +1,99 @@
 mod client;
 mod util;
+use futures::{future, stream::Stream, Async, Future, Poll};
+use http::{Request, Response};
+use hyper::{
+    client::{HttpConnector, ResponseFuture},
+    Body, Client as HyperClient, body::Payload
+};
+use bytes::Bytes;
+use tower_service::Service;
+use tower_retry::Policy;
 
 pub use self::client::{Connect, Connection};
+
+struct Client<C = HttpConnector, B = Body> {
+    inner: HyperClient<C, B>
+}
+
+impl Default for Client<HttpConnector, Body> {
+    fn default() -> Self {
+        Self {
+            inner: HyperClient::new()
+        }
+    }
+}
+
+impl <B>Service<Request<B>> for Client<B>
+where
+    Body: From<B>,
+    B: hyper::client::connect::Connect,
```

Comment:
this seems a bit odd, why would B implement both connect and Payload?

Reply:
That's a good question—some weird type error came up? I'll investigate more tomorrow.

Reply:
what _worked_ was effectively copying hyper's client generics. I'm not sure how to fit `ConnectService` into this place, however. Is that why you implemented Service on `Connection` initially?

Reply:
Yeah, because in `Connection` it accepts a `AsyncRead + AsyncWrite` which is what the hyper `Connect` trait does. The issue is that for `Connection` we implement `ConnectService` for any `Connect` but I don't think the other way around will work. The idea around `ConnectService` is to provide out of the box ways to generate those transports like provide dns resolution and then tls etc.  Though to be honest here it's not that big of a deal since we are using the thick hyper client whereas the connect stuff is really needed for the lower level components.